### PR TITLE
Changelog entry for v0.3.1 release

### PR DESCRIPTION
**Pull request by willschlitzer**
**Description of proposed changes**

**Preview**: https://pygmt-git-v031-release-gmt.vercel.app/changes.html#changelog

Previous release at #887

Changelog made by following the instructions at https://github.com/GenericMappingTools/pygmt/blob/master/MAINTENANCE.md#updating-the-changelog. The changelog itself includes the following sections:

- New features
- Enhancements
- Documentation
- Bug fixes
- Maintenance
- List of Contributors

TODO: 

- [x] Add Zenodo DOI link once it's reserved.
- [x] Add a new entry in doc/_static/version_switch.js for documentation switcher
- [x] Update citation information https://github.com/GenericMappingTools/pygmt#citing-pygmt
- [x] Set release date

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If adding new functionality, add an example to docstrings or tutorials.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

### CODE REVIEW

**Comment by willschlitzer** [comment]
@weiji14 @seisman It looks like the auto-generated changelog is already formatting the associated links; do I still need to run ``sed -i.bak -E 's$\(#([0-9]*)\)$(`#\1 <https://github.com/GenericMappingTools/pygmt/pull/\1>`__)$g' changes.rst``?

File: doc/changes.md
Comment (core-man) [inline doc/changes.md:31]:
```suggestion
* Add gallery example showing different polar projection use cases ([#955](https://github.com/GenericMappingTools/pygmt/pull/955))
```

**Comment by seisman** [comment]
> It looks like the auto-generated changelog is already formatting the associated links; do I still need to run `` sed -i.bak -E 's$\(#([0-9]*)\)$(`#\1 <https://github.com/GenericMappingTools/pygmt/pull/\1>`__)$g' changes.rst ``?

No need.

File: README.rst
Comment (seisman) [inline README.rst:239]:
```suggestion
0.3.x   >=6.1.1    >=3.7
```

File: doc/changes.md
Comment (seisman) [inline doc/changes.md:64]: There are a few contributors that are not included here, why?

Comment (willschlitzer) [reply doc/changes.md:64]: I included everyone in AUTHORS.md; should I have included those who haven't added themselves?

Comment (weiji14) [reply doc/changes.md:64]: Yes, please include everyone. There should be a git log command to check who commited between v0.3.0 and v0.3.1

Comment (seisman) [reply doc/changes.md:64]: See here. We want to include every one. https://github.com/GenericMappingTools/pygmt/blob/master/AUTHORSHIP.md#changelog-for-each-release

File: doc/changes.md
Comment (seisman) [inline doc/changes.md:38]: Move all entries that "add a gallery example" together:
```suggestion
* Add a gallery example showing different line front styles ([#1022](https://github.com/GenericMappingTools/pygmt/pull/1022))
* Add a gallery example for a double y-axes graph ([#1019](https://github.com/GenericMappingTools/pygmt/pull/1019))
* Add a gallery example of inset map showing a rectangle region ([#1020](https://github.com/GenericMappingTools/pygmt/pull/1020))
* Add a gallery example to show coloring of points by categories ([#1006](https://github.com/GenericMappingTools/pygmt/pull/1006))
* Add gallery example showing different polar projection use cases ([#955](https://github.com/GenericMappingTools/pygmt/pull/955))
* Add a gallery example for plotting Cartesian, circular and geographic vectors ([#950](https://github.com/GenericMappingTools/pygmt/pull/950))
* Add a gallery example showing different vector heads and tails ([#890](https://github.com/GenericMappingTools/pygmt/pull/890))
* Reorganize gallery examples ([#995](https://github.com/GenericMappingTools/pygmt/pull/995))
* Add underscore guideline to CONTRIBUTING.md ([#1034](https://github.com/GenericMappingTools/pygmt/pull/1034))
* Add instructions to upgrade installed PyGMT version ([#1029](https://github.com/GenericMappingTools/pygmt/pull/1029))
* Improve the docstring of the pygmt package ([#1016](https://github.com/GenericMappingTools/pygmt/pull/1016))
* Add common alias coltypes (-f) for specifying i/o data types ([#994](https://github.com/GenericMappingTools/pygmt/pull/994))
* Expand documentation linking in CONTRIBUTING.md ([#802](https://github.com/GenericMappingTools/pygmt/pull/802))
* Write changelog in markdown using MyST ([#941](https://github.com/GenericMappingTools/pygmt/pull/941))
* Update web font to Atkinson Hyperlegible ([#938](https://github.com/GenericMappingTools/pygmt/pull/938))
* Improve the gallery example for datetime inputs ([#919](https://github.com/GenericMappingTools/pygmt/pull/919))
```

File: doc/changes.md
Comment (weiji14) [inline doc/changes.md:7]: I'll let people have a go at choosing the highlights this time :)

```suggestion
### Highlights

*🎉 **Multiple bug fixes and a reorganized gallery** 🎉
*
*
```

File: doc/changes.md
Comment (willschlitzer) [reply doc/changes.md]: I added @tawandamoyo and @aliciaha1997, which I think completes the list. Both @michaelgrund and @core-man both look like they had repeat entries when I ran `git shortlog HEAD...v0.3.0 -sne`, and I figured I wouldn't include dependabot (this will definitely be a problem for me if and when there is a robot uprising).

Comment (weiji14) [reply doc/changes.md]: > I figured I wouldn't include dependabot (this will definitely be a problem for me if and when there is a robot uprising).

There should a way to prevent the bot uprising soon with https://github.com/release-drafter/release-drafter/issues/569 :slightly_smiling_face:

File: doc/changes.md
Comment (weiji14) [inline doc/changes.md:11]: Still need a couple of highlight items here team! What are the key changes to the documentation we want to highlight to our users? Do we want to mention that PyGMT v0.3.1 is the last release to support GMT 6.1.1 (and that the next version will require GMT 6.2.0)?

Comment (core-man) [reply doc/changes.md:11]: For the documentation, I'd like to recommend [Cartesian, circular, and geographic vectors](https://www.pygmt.org/dev/gallery/lines/vectors.html#sphx-glr-gallery-lines-vectors-py)

> Do we want to mention that PyGMT v0.3.1 is the last release to support GMT 6.1.1 (and that the next version will require GMT 6.2.0)?

Sounds good.

Comment (willschlitzer) [reply doc/changes.md:11]: Works for me! Updating now.

File: doc/changes.md
Comment (weiji14) [inline doc/changes.md:11]:
```suggestion
* This is the last PyGMT version to support GMT 6.1.1; the next release will require GMT 6.2
```

File: doc/changes.md
Comment (weiji14) [inline doc/changes.md:32]: Take this away since it's in the highlights.
```suggestion
```

File: doc/changes.md
Comment (weiji14) [inline doc/changes.md:10]: Add link to the PR here.
```suggestion
* New gallery example for Cartesian, circular, and geographic vectors ([#950](https://github.com/GenericMappingTools/pygmt/pull/950))
```

File: doc/changes.md
Comment (weiji14) [inline doc/changes.md:9]: I think we can add one more highlight (usually we have 3 in total). Any more suggestions?

Comment (core-man) [reply doc/changes.md]: Then I'd like to recommend another gallery example: [Vector heads and tails](https://www.pygmt.org/dev/gallery/lines/vector_heads_tails.html#sphx-glr-gallery-lines-vector-heads-tails-py). It is cool and useful.

File: doc/changes.md
Comment (weiji14) [inline doc/changes.md:10]: > Then I'd like to recommend another gallery example: [Vector heads and tails](https://www.pygmt.org/dev/gallery/lines/vector_heads_tails.html#sphx-glr-gallery-lines-vector-heads-tails-py). It is cool and useful. How do you think ?@weiji14 @willschlitzer

Yep, actually #950 and #890 are part of the vectors issue at #892 so maybe we could merge it into one bullet point?

For another point, how about mentioning the gallery reorganization #995? There was quite a bit of work (and debate) that went into that!

Comment (core-man) [reply doc/changes.md:10]: > Yep, actually #950 and #890 are part of the vectors issue at #892 so maybe we could merge it into one bullet point?

Sounds good.

> For another point, how about mentioning the gallery reorganization #995? There was quite a bit of work (and debate) that went into that!

I agree, but the first line says:

> 🎉 **Multiple bug fixes and a reorganized gallery** 🎉

So can we still highlight it again?

Comment (willschlitzer) [reply doc/changes.md]: Added in 444ef30

File: doc/changes.md
Comment (weiji14) [inline doc/changes.md:5]: Almost forgot to change the DOI number!

```suggestion
[![Digital Object Identifier for PyGMT v0.3.1](https://zenodo.org/badge/DOI/10.5281/zenodo.4592991.svg)](https://doi.org/10.5281/zenodo.4592991)
```

File: doc/changes.md
Comment (weiji14) [reply doc/changes.md]: > I agree, but the first line says:
> 
> > tada **Multiple bug fixes and a reorganized gallery** tada
> 
> So can we still highlight it again?

Or we can change that first line to say ":tada: Multiple bug fixes and an improved gallery :tada:" :shrug:

File: doc/changes.md
Comment (willschlitzer) [reply doc/changes.md]: 🤦🏻‍♂️

File: doc/changes.md
Comment (weiji14) [reply doc/changes.md]: Alternatively, we can mention the Twitter account :bird:https://twitter.com/gmt_software

Comment (core-man) [reply doc/changes.md]: > Or we can change that first line to say " Multiple bug fixes and an improved gallery "

That's great. Will update https://hackmd.io/@pygmt/ryt7DI5md when other maintainers approve this PR.

Comment (weiji14) [reply doc/changes.md]: Great, thanks for starting that changelog @core-man! I've added a nice link so using https://hackmd.io/@pygmt/v0-3-1_announcement now:

image

File: doc/changes.md
Comment (weiji14) [inline doc/changes.md:12]:
```suggestion
* 🎉 **Multiple bug fixes and an improved gallery** 🎉
* Reorganized gallery examples into new categories ([#995](https://github.com/GenericMappingTools/pygmt/pull/995))
* Added gallery examples for plotting vectors ([#950](https://github.com/GenericMappingTools/pygmt/pull/950), [#890](https://github.com/GenericMappingTools/pygmt/pull/890))
* Last version to support GMT 6.1.1, future PyGMT versions will require GMT 6.2.0 or newer
```

File: doc/changes.md
Comment (michaelgrund) [reply doc/changes.md]: From my point of view, the most important changes are covered 😉.

File: doc/changes.md
Comment (weiji14) [inline doc/changes.md:61]: @core-man has more commits than me so should move up the spot :)

```suggestion
* [Jiayuan Yao](https://github.com/core-man)
* [Wei Ji Leong](https://github.com/weiji14)
```

File: doc/changes.md
Comment (weiji14) [inline doc/changes.md:64]: All three of you actually have 9 commits (@willschlitzer has 8, +1 after merging this PR), but `git shortlog HEAD...v0.3.0 -sne` sorts things alphabetically so...

```suggestion
* [Meghan Jones](https://github.com/meghanrjones)
* [Michael Grund](https://github.com/michaelgrund)
* [Will Schlitzer](https://github.com/willschlitzer)
```

**Review by weiji14 — COMMENTED** [review]
Hopefully the last nitpick, sorting the contributor list by number of commits, and tied ones are sorted alphabetically as per `git shortlog HEAD...v0.3.0 -sne` (on the master branch).

```
   25  Dongdong Tian
    19  Yao Jiayuan
    14  Wei Ji
     9  Meghan Jones
     8  Michael Grund
     8  Will Schlitzer
     3  dependabot[bot]
     2  Liam Toney
     2  kmaterna
     1  Alicia Ngoc Diep Ha
     1  Michael Grund
     1  Tawanda
     1  Yao Jiayuan (姚家园)
```

:wave: Everyone else, preview things at https://pygmt-git-v031-release-gmt.vercel.app/changes.html#release-v0-3-1-2021-03-14 to see if there's anything amiss.

File: doc/changes.md
Comment (willschlitzer) [reply doc/changes.md]: I realize now why my order was different was that I didn't switch to the master branch; I didn't think I had so many commits!

**Comment by core-man** [comment]
Following @weiji14 , I checked [Citing PyGMT](https://pygmt-git-v031-release-gmt.vercel.app/index.html#citing-pygmt) and authorship on Zenodo archives of releases at [AUTHORSHIP.md](https://github.com/GenericMappingTools/pygmt/blob/master/AUTHORSHIP.md):

> The order of authors will be defined by the number of commits to the repository (git shortlog -sne). The order can also be changed on a case-by-case basis.

```
$ git shortlog -sne
   256  Leonardo Uieda <leouieda@gmail.com>
   225  Dongdong Tian <seisman.info@gmail.com>
   112  Wei Ji <23487320+weiji14@users.noreply.github.com>
    63  Will Schlitzer <schlitzer90@gmail.com>
    19  Yao Jiayuan <coreman.seism@gmail.com>
    16  Michael Grund <michael_grund@gmx.de>
    15  Liam Toney <38269494+liamtoney@users.noreply.github.com>
    12  Meghan Jones <meghanj@hawaii.edu>
     9  dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
     3  kmaterna <kathrynmaterna@gmail.com>
     2  Andrey Shmakov <akshmakov@gmail.com>
     2  Conor Bacon <conor.bacon@gmail.com>
     2  Tyler Newton <54011699+tjnewton@users.noreply.github.com>
     1  Alicia Ngoc Diep Ha <58379930+aliciaha1997@users.noreply.github.com>
     1  Brook <btozer@ucsd.edu>
     1  Claudio Satriano <satriano@gmail.com>
     1  Cody Woodson <dovacody@gmail.com>
     1  Malte Ziebarth <malte.ziebarth@fmvkb.de>
     1  Mark Wieczorek <mark.a.wieczorek@gmail.com>
     1  Michael Grund <23025878+michaelgrund@users.noreply.github.com>
     1  Philipp Loose <37411699+xibalba01@users.noreply.github.com>
     1  Tawanda <moyotawanda@gmail.com>
     1  Yao Jiayuan (姚家园) <coreman.seism@gmail.com>
     1  carocamargo <47150926+carocamargo@users.noreply.github.com>
```

It seems that we also need to update the authors for the citation. Am I right?
```
Uieda, Leonardo and
Tian, Dongdong and
Leong, Wei Ji and
Schlitzer, William and
Yao, Jiayuan and
Grund, Michael and
Toney, Liam and
Jones, Meghan and
Materna, Kathryn and
Newton, Tyler and
Ziebarth, Malte and
Wessel, Paul
```

File: README.rst
Comment (core-man) [inline README.rst:164]:
```suggestion
                      Schlitzer, William and
                      Yao, Jiayuan and
                      Grund, Michael and
                      Toney, Liam and
```

**Comment by core-man** [comment]
I quickly reviewed the changelog again and have no more comments.

File: README.rst
Comment (willschlitzer) [reply README.rst:164]: I had based this off number of commits yesterday; I'll put  
`` Toney, Liam and
Schlitzer, William and
Yao, Jiayuan and
Grund, Michael and
``

In the v0.3.0 release we put @liamtoney up higher because of his ROSES video, which may not count towards his commits but has significantly enhanced the project. Ping @GenericMappingTools/python for input, but I think he should remain where he is in the order because he is spearheading the SciPy session.

Comment (weiji14) [reply README.rst]: > Following @weiji14 , I checked [Citing PyGMT](https://pygmt-git-v031-release-gmt.vercel.app/index.html#citing-pygmt) and authorship on Zenodo archives of releases at [AUTHORSHIP.md](https://github.com/GenericMappingTools/pygmt/blob/master/AUTHORSHIP.md):
> 
> > The order of authors will be defined by the number of commits to the repository (git shortlog -sne). The order can also be changed on a case-by-case basis.

Yes, on a first order, we sort things by the number of commits. But pay attention to the next sentence "The order can also be changed on a case-by-case basis". The idea is that contributions are not just about commits but also the other things people do. For example, Liam put a lot of work into the ROSES tutorial (the video on the front page) and Meghan has done lots of things behind the scenes such as funding applications and setting up accounts.

I don't want to get too fired up about name ordering, but it's really up to you all to negotiate what you think is best. Just stating that there are other factors to consider :)

Comment (willschlitzer) [reply README.rst]: Changes pushed in 8fef314

Comment (core-man) [reply README.rst]: Good. That's reasonable.

Comment (willschlitzer) [reply README.rst]: >  Actually, I begin to know PyGMT from his ROSES tutorial. 

Me too! Thanks Liam!

**Review by weiji14 — APPROVED** [review]
Let's get this merged! Zenodo is at the ready too!

image

**Comment by willschlitzer** [comment]
> Let's get this merged! Zenodo is at the ready too!
> 
> image

I can't "Squash and merge" because of the required tests; can you disable them/merge this PR?

**Comment by weiji14** [comment]
Yep, I can merge this one in (unless someone has any last minute issues). About time we give you admin permissions @willschlitzer :smile:

File: README.rst
Comment (michaelgrund) [reply README.rst]: Everythin fine for me 😉.

**Pull request merged**
